### PR TITLE
WIP: major eslint updates (with eslint-config-standard@next) for evaluation only

### PR DESCRIPTION
**Pull request by brodybits**
### Platforms affected

iOS

### What does this PR do?

WIP for evaluation & discussion only (with an alt-wip branch name):

- major eslint updates, using eslint-config-standard@next _(^12.0.0-alpha.0) as needed to resolve `[ESLINT_LEGACY_OBJECT_REST_SPREAD]` DeprecationWarning (<https://github.com/standard/eslint-config-standard/issues/123#issuecomment-404772247>) when using eslint@5_, taken from PR #382 

cannot be backported to 4.5.x since it requires minimum Node.js 6

_Alternative using eslint@4, not using alpha eslint-config-standard@next version, is in #389._

### What testing has been done on this change?

- [x] check that `npm test` items pass on Travis CI

### Checklist

- ~~[Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database~~
- ~~Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.~~
- ~~Added automated test coverage as appropriate for this change.~~

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,13 +32,13 @@
...
-    "eslint": "^4.0.0",
-    "eslint-config-semistandard": "^11.0.0",
-    "eslint-config-standard": "^10.2.1",
-    "eslint-plugin-import": "^2.7.0",
-    "eslint-plugin-node": "^5.1.0",
-    "eslint-plugin-promise": "^3.5.0",
-    "eslint-plugin-standard": "^3.0.1",
+    "eslint": "^5.2.0",
+    "eslint-config-semistandard": "^12.0.1",
+    "eslint-config-standard": "^12.0.0-alpha.0",
```
Comment (erisu) [inline package.json]: Is there a reason that this one has to be on `@next`? 
Could it be `eslint-config-standard@^11.0.0` as seen in the other evaluation PR #389 while the rest were updated?

I don't see any problems with updating all of them.

Comment (brodybits) [reply package.json]: eslint-config-standard@next was needed to resolve an ugly warning message from eslint@5, details are now in the description

Comment (brodybits) [reply package.json]: FYI `^12.0.0-alpha.0` means that we would use the -alpha version until the publish a newer version, such as -beta, or finally publish 12.0.0 version

**Comment by brodybits** [comment]
To clarify the general question: I would generally favor upgrading to "latest" version of dependencies and devDependencies whenever possible when targeting upcoming major release. The problem is that using eslint@latest would lead to the ugly warning as discussed in the updated description and review comments unless we would use eslint-config-standard@next (eslint-config-standard@^12.0.0-alpha.0). We are basically stuck waiting for eslint-config-standard@12.0.0 to be released as discussed in standard/eslint-config-standard#123.

I think we should not introduce "@next" dependency version unless absolutely necessary for some reason.

**Comment by shazron** [comment]
Is this still WIP?

**Comment by erisu** [comment]
Closing in favor https://github.com/apache/cordova-ios/pull/496 which is now merged.

**Pull request closed without merging**